test/cli_test.c

```diff
@@ -0,0 +1,160 @@
+    char *argv[] = {"./yobemag", "-s", "0"};
+    int argc     = 3;
+
+    CLIArguments cli_args;
+    cli_parse(&cli_args, argc, argv);
+}
+
+Test(cli, cli_too_many_args, .exit_code = EXIT_FAILURE) {
+    char *argv[] = {"./yobemag", "-l", "4", "../build/yobemag.gb", "test_param"};
+    int argc     = 5;
+
+    CLIArguments cli_args;
+    cli_parse(&cli_args, argc, argv);
+}
+
+Test(cli, cli_no_rom_path, .exit_code = EXIT_FAILURE) {
+    char *argv[] = {"./yobemag", "-l", "4"};
+    int argc     = 3;
+
+    CLIArguments cli_args;
+    cli_parse(&cli_args, argc, argv);
+}
+
+Test(cli, cli_no_log_lvl_no_path, .exit_code = EXIT_FAILURE) {
+    char *argv[] = {"./yobemag", "-l"};
+    int argc     = 2;
+
+    CLIArguments cli_args;
+    cli_parse(&cli_args, argc, argv);
+}
+
+Test(cli, cli_no_log_lvl, .exit_code = EXIT_FAILURE) {
+    char *argv[] = {"./yobemag", "-l", "../build/yobemag.gb"};
+    int argc     = 3;
+
+    CLIArguments cli_args;
+    cli_parse(&cli_args, argc, argv);
+}
+
+Test(cli, cli_not_finished, .exit_code = EXIT_FAILURE) {
+    char *argv[] = {"./yobemag", "-l", "1abc"};
+    int argc     = 3;
+
+    CLIArguments cli_args;
+    cli_parse(&cli_args, argc, argv);
+}
+
+Test(cli, cli_gt_than_long_max, .exit_code = EXIT_FAILURE) {
+    int const n = snprintf(NULL, 0, "%li", LONG_MAX);
+    char buf[n + 2];
+    int c  = snprintf(buf, n + 1, "%li", LONG_MAX);
+    buf[n] = '9';
+
+    char *argv[] = {"./yobemag", "-l", buf};
+    int argc     = 3;
+
+    CLIArguments cli_args;
+    cli_parse(&cli_args, argc, argv);
+}
+
+Test(cli, cli_lt_than_long_min, .exit_code = EXIT_FAILURE) {
+    int const n = snprintf(NULL, 0, "%li", LONG_MIN);
+    char buf[n + 2];
+    int c  = snprintf(buf, n + 1, "%li", LONG_MIN);
+    buf[n] = '9';
+
+    char *argv[] = {"./yobemag", "-l", buf};
+    int argc     = 3;
+
+    CLIArguments cli_args;
+    cli_parse(&cli_args, argc, argv);
+}
+
+Test(cli, cli_gt_than_int_max, .exit_code = EXIT_FAILURE) {
+    int const n = snprintf(NULL, 0, "%li", LONG_MAX);
+    char buf[n + 2];
+    int c = snprintf(buf, n + 1, "%li", LONG_MAX);
+
+    char *argv[] = {"./yobemag", "-l", buf};
+    int argc     = 3;
+
+    CLIArguments cli_args;
+    cli_parse(&cli_args, argc, argv);
+}
+
+Test(cli, cli_lt_than_int_min, .exit_code = EXIT_FAILURE) {
+    int const n = snprintf(NULL, 0, "%li", LONG_MIN);
+    char buf[n + 2];
+    int c = snprintf(buf, n + 1, "%li", LONG_MIN);
+
+    char *argv[] = {"./yobemag", "-l", buf};
+    int argc     = 3;
+
+    CLIArguments cli_args;
+    cli_parse(&cli_args, argc, argv);
+}
+
+Test(cli, cli_correct_with_log_lvl, .exit_code = EXIT_SUCCESS) {
+    char *argv[] = {"./yobemag", "-l", "2", "../build/yobemag.gb"};
+    int argc     = 4;
+
+    CLIArguments cli_args;
+    cli_parse(&cli_args, argc, argv);
+
+    uint8_t actual_log_lvl   = cli_args.logging_level;
+    uint8_t expected_log_lvl = 2;
+
+    // resolves discarded-qualifiers warning without removing const from CLIArguments
```

Comment:
What does this comment mean? 🤔

Reply:
if you try to assert this string directly as you proposed earlier, the "Wdiscarded-qualifiers" warning complains with "initialization discards ‘const’ qualifier from pointer target type".  
This is not solved by using `const` as the warning implies, therefore, I worked around it. 
I have assumed it to be a bug in the compiler as nothing seemed to fix it, except removing the const qualifier inside the CLIArguments struct

Reply:
I see 🤔 I think the `eq(...)` is probably not supposed to be used with string literals then, but only `char*`

This could work instead:
```C
const char* expected = "../build/yobemag.gb";
cr_assert_str_eq(cli_args.rom_path, expected);
```

Reply:
This works! Thanks